app/src/main/kotlin/com/wire/android/util/permission/TakePictureRequestFlow.kt

```diff
@@ -8,30 +8,38 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
 import androidx.compose.ui.platform.LocalContext
 
+/**
+ * Flow that will launch the camera for taking a photo.
+ * This will handle the permissions request in case there is no permission granted for the camera.
+ *
+ * @param onPictureTaken action that will be executed for Camera's [ActivityResultContract]
+ * @param onPermissionDenied action to be executed when the permissions is denied
+ * @param targetPictureFileUri target file where the media will be stored
+ */
 @Composable
 fun rememberTakePictureFlow(
-    shouldPersistUri: (Boolean) -> Unit,
+    onPictureTaken: (Boolean) -> Unit,
```

Comment:
same here maybe we could use some kind of generic activity result wrapper to give this boolean more meaning

Reply:
I think this could stay as it is right now, in case there is some confusion when using this in other possible places, maybe then we can revisit it. Right now, there are only 2 flows that are expected to be used, avatar and sharing an image.

But I made a small improvement at least in terms of doc, I added it at the top of the composable just in case :D